### PR TITLE
Link unsigned and rational FHE types

### PR DESCRIPTION
**Pull request by samtay**

### CODE REVIEW

File: sunscreen/tests/linked.rs [hunk 1]
```diff
@@ -162,6 +163,150 @@ mod linked_tests {
...
+    fn compare_rational<F: FieldSpec>(#[linked] x: BfvRational<F>, #[linked] y: BfvRational<F>) {
+        let (x_n, x_d) = x.into_field_elem();
+        let (y_n, y_d) = y.into_field_elem();
+        let x = x_n * y_d;
+        let y = y_n * x_d;
+        x.constrain_le_bounded(y, 64)
+    }
+
+    #[test]
+    #[ignore = "bug in linked input multplication"]
```
Comment (samtay) [inline sunscreen/tests/linked.rs:268]: This issue was discovered in these tests, but it is preexisting and independent of the work in this PR. Will be tackled next week.

File: sunscreen_runtime/src/linked.rs [hunk 2]
```diff
@@ -89,6 +89,7 @@ fn new_single_party_with_shared_generators(
         if left_side {
             g[index] = *gen;
             left_side = false;
+            // TODO this overflows if shared arguments are provided with no constraints
```
Comment (samtay) [inline sunscreen_runtime/src/linked.rs:92]: Similar to above, something that needs to be tackled when we shore up the shared commitments, but unrelated to this PR.

File: sunscreen/tests/linked.rs (hunk 1 above)
Comment (ryanorendorff) [reply sunscreen/tests/linked.rs:268]: Ah I see! I bet the multiplication is throwing off the assumed generator order by adding a new input prior to where the link component thinks the multipliers will be.

File: sunscreen_runtime/src/linked.rs (hunk 2 above)
Comment (ryanorendorff) [reply sunscreen_runtime/src/linked.rs:92]: As in the ZKP takes in a shared input but does nothing with them?

**Review by ryanorendorff — APPROVED** [review]
lgmt!

File: sunscreen_runtime/src/linked.rs (hunk 2 above)
Comment (samtay) [reply sunscreen_runtime/src/linked.rs:92]: Yea exactly

**Pull request merged**